src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -192,6 +192,29 @@ public struct ExceptionInformationArray
         }
 
         private static ulong s_crashingThreadId;
+        private static CrashInfo s_crashInfo;
+        private static int s_crashInfoPresent;
+
+        internal static CrashInfo SerializeCrashInfo(RhFailFastReason reason, string? message, Exception? exception)
+        {
+            int previousState = Interlocked.CompareExchange(ref s_crashInfoPresent, 1, 0);
+            if (previousState == 0)
+            {
+                s_crashInfoPresent = 1;
+
+                CrashInfo crashInfo = new();
+
+                crashInfo.Open(reason, Thread.CurrentOSThreadId, message ?? GetStringForFailFastReason(reason));
+                if (exception != null)
+                {
+                    crashInfo.WriteException(exception);
+                }
+                crashInfo.Close();
+                s_crashInfo = crashInfo;
+            }
+
+            return s_crashInfo;
```

Comment:
```suggestion
                s_triageBufferAddress = crashInfo.TriageBufferAddress;
                s_triageBufferSize = crashInfo.TriageBufferSize;

                s_crashInfoPresent = 1;
            }
```
We do not need to pass the full CrashInfo writer around. It is enough to just keep the triage buffer.